### ISSUE TITLE
Ability to generate summary using pattern match against tags and annotations

### ISSUE THREAD

**Issue by CNG** [GothenburgBitFactory/timewarrior#339]
I would like to compute the time spent cumulatively on tasks within certain projects. I currently have many intervals in Timewarrior with tags based on ticket numbers from my project management software, such as `AA-aa-123`, `BB-aa-123` and `BB-bb-123`. To compute the total time spent on project BB-aa, I currently need to do something like `timew summary :year | grep 'BB-aa'` and transform the output to put into a spreadsheet where I can sum the timestamps. This is tedious and painful.

Probably this would be easier if I tagged everything like `timew start BB BB-aa BB-aa-123` and then later I could do a `timew summary BB-aa`. I don't love the idea of having redundant data in tags, which would not be necessary if I could do `timew summary BB-aa-*` or `timew summary /BB-aa-\+*/`. Note it's also difficult to retroactively change my tagging strategy as I discover problems with being able to later summarize that data since I can't easy go back and change tons of intervals to, for example, add those `BB` and `BB-aa` tags. That would require both the pattern searching I'm describing here as well as bulk modifying I described in https://github.com/GothenburgBitFactory/timewarrior/issues/338.

**Comment by lauft** [comment]
See also #64 and #209

**Comment by lauft** [comment]
I agree that pattern matching/boolean logic for tags would be a powerful addition to Timewarrior, not only for the summary command but in general.

**Comment by tobixen** [comment]
I may have some similar use-cases, and have published some of my scripts at https://github.com/tobixen/timewarrior-tools - this includes a reporting script that accepts regexps for tags to include and exclude.  I have tags that includes github issue numbers and similar IDs for other tracking systems.  In my use-case I also have a script that does lookup of the ID in a database and adds tags based on what it finds (the script doing the lookups is not included in the collection above, the script for adding tags is included).
